Sound/Tidal/EspGrid.hs

```diff
@@ -0,0 +1,114 @@
+module Sound.Tidal.EspGrid where
+
+import Control.Concurrent.MVar
+import Control.Concurrent
+import Control.Monad (forever)
+import Control.Monad.Loops (iterateM_)
+import Sound.OSC.FD
+import Data.Time.Clock
+import Data.Time.Clock.POSIX
+import Data.Time.Calendar (fromGregorian)
+
+import Sound.Tidal.Tempo
+import Sound.Tidal.Time as T
+import Sound.Tidal.Stream
+import Sound.Tidal.Dirt
+import Sound.Tidal.Transition (transition)
+import Sound.Tidal.Pattern (silence)
+
+parseEspTempo :: [Datum] -> Maybe Tempo
+parseEspTempo d = do

```

Comment:
True! If you wanted to support extra args you'd probably say

```parseEspTempo (on' : bpm' : t1' : t2' : n' : _) = do```

The reasons for this style are:
  - Easier not to mix up arguments and numbers, and you can just call them by their names
  - ```(!!)``` is somewhat frowned upon, as it's a partial function. The pattern-matching is partial too, but you get a warning from the compiler about it